pkg/common/server/config.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	k8s "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+	componentbaseconfig "k8s.io/component-base/config"
+	"os"
+
+	"github.com/gardener/gardener/pkg/logger"
+)
+
+// RestConfigFromClientConnectionConfiguration creates a *rest.Config from a componentbaseconfig.ClientConnectionConfiguration & the configured kubeconfig
+func RestConfigFromClientConnectionConfiguration(cfg componentbaseconfig.ClientConnectionConfiguration) (*rest.Config, error) {
+	restConfig, err := clientcmd.BuildConfigFromFlags("", cfg.Kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	restConfig.Burst = int(cfg.Burst)
+	restConfig.QPS = cfg.QPS
+	restConfig.AcceptContentTypes = cfg.AcceptContentTypes
+	restConfig.ContentType = cfg.ContentType
+	return restConfig, nil
+}
+
+// CreateRecorder creates a record.EventRecorder that is not limited to a namespace having a specific eventSourceName
+func CreateRecorder(kubeClient k8s.Interface, eventSourceName string) record.EventRecorder {
```

Comment:
As said earlier - this is unrelated to the (HTTP/S server) package. Move it somewhere else.